### PR TITLE
Add authentication and authorization

### PR DESCRIPTION
**Pull request by mskab**

### CODE REVIEW

**Review by mikaserova — COMMENTED** [review]
Lgtm, but you need to add auth to endpoints

**Comment by mskab** [comment]
> Lgtm, but you need to add auth to endpoints

I updated it

File: api/db/repository/user.py [hunk 1]
```diff
@@ -66,6 +68,11 @@ def update(_db: Session, user_id: int, user: UserUpdate):
             update_user_encoded["password"]
         )
 
+    if "is_active" in update_user_encoded and isinstance(
```
Comment (mikaserova) [inline api/db/repository/user.py]: `
if update_user_encoded.get("is_active") is not None:
`

File: api/routes/user.py [hunk 2]
```diff
@@ -20,39 +28,83 @@ def create_user(req_user: UserCreate, db: Session = default_session):
...
+    db: Session = default_session,
+    Auth: AuthJWT = default_authJWT,
+):
     """
     Delete the User with the given ID
     """
+    auth.is_only_admin_permitted(db, Auth)
     user.delete(db, user_id)
 
     return "User deleted successfully"
```
Comment (mikaserova) [inline api/routes/user.py]: You can use smth like this for cases where you return only status message:
```
class StatusResponse(BaseResponse):
    success: bool
    message: Optional[str] = None
```

**Review by mikaserova — APPROVED** [review]

File: api/db/repository/user.py (hunk 1 above)
Comment (mskab) [reply api/db/repository/user.py]: I updated it

File: api/routes/user.py (hunk 2 above)
Comment (mskab) [reply api/routes/user.py]: I added it

**Review by mikaserova — APPROVED** [review]

**Pull request merged**